src/java/main/org/apache/zookeeper/server/auth/SASLAuthenticationProvider.java

```diff
@@ -38,11 +38,6 @@ public String getScheme() {
     }
 
     public boolean matches(String id,String aclExpr) {
-        if (System.getProperty("zookeeper.superUser") != null) {
-            if (id.equals(System.getProperty("zookeeper.superUser")) || id.equals(aclExpr)) {
-              return true;
-            }
-        }
         if ((id.equals("super") || id.equals(aclExpr))) {
```

Comment:
I think we should remove (id.equals("super") also. Lets have only one way being super user.